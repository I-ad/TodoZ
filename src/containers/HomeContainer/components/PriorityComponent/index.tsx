// import * as PropTypes from 'prop-types';
import * as React from 'react';

interface IProps {
  value: string;
  error: string;
  isTouched: boolean;
  isSubmitted: boolean;

  onChange(e: any): void;
}

class PriorityComponent extends React.Component<IProps> {
  public shouldComponentUpdate(nextProps: Readonly<IProps>): boolean {
    return nextProps.value !== this.props.value || nextProps.isSubmitted !== this.props.isSubmitted;
  }

  public render(): JSX.Element {
    const { value, onChange, error, isSubmitted, isTouched } = this.props;
    return (
      <React.Fragment>
        <label>Priority</label>
        <input name="priority" type="radio" value="low" onChange={onChange} checked={value === 'low'} /> low
        <input name="priority" type="radio" value="mid" onChange={onChange} checked={value === 'mid'} /> mid
        <input name="priority" type="radio" value="high" onChange={onChange} checked={value === 'high'} /> high
        <p>{(isSubmitted || isTouched) && error}</p>
      </React.Fragment>
    );
  }
}

export default PriorityComponent;
