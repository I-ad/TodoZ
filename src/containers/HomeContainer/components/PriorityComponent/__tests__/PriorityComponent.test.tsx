import * as Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import PriorityComponent from '../index';

it('should render correctly', () => {
  const onChange = () => null;
  const wrapper = Enzyme.shallow(
    <PriorityComponent
      value="low"
      error="error"
      isTouched={true}
      isSubmitted={true}
      onChange={onChange}
    />);
  expect(toJson(wrapper)).toMatchSnapshot();
}); 
