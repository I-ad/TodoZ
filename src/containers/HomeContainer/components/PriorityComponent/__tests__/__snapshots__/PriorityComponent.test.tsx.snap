// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<React.Fragment>
  <label>
    Priority
  </label>
  <input
    checked={true}
    name="priority"
    onChange={[Function]}
    type="radio"
    value="low"
  />
   low
  <input
    checked={false}
    name="priority"
    onChange={[Function]}
    type="radio"
    value="mid"
  />
   mid
  <input
    checked={false}
    name="priority"
    onChange={[Function]}
    type="radio"
    value="high"
  />
   high
  <p>
    error
  </p>
</React.Fragment>
`;
